sympy/functions/special/delta_functions.py

```diff
@@ -67,6 +67,8 @@ def eval(cls, arg, k=0):
             return S.NaN
         if arg.is_positive or arg.is_negative:
             return S.Zero
+        if fuzzy_not(im(arg).is_zero):
+            raise ValueError("Function defined only for Real Values. Complex part: %s  found in %s ." % (repr(im(arg)), repr(arg)) )
```

Comment:
This is a nicer way to deal with long error messages and to use the new formatting method.:

```python
    msg = "Function defined only for Real Values. Complex part: {} found in {}."
    raise ValueError(msg.format(repr(im(arg)), repr(arg))
```